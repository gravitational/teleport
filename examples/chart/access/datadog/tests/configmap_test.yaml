suite: Test configmap
templates:
  - configmap.yaml
tests:
  - it: should match the snapshot
    set:
      teleport:
        address: teleport.example.com:1234
      datadog:
        apiEndpoint: https://api.datadoghq.com
        apiKey: test-api-key
        applicationKey: test-application-key
        fallbackRecipient: admin@example.com
        severity: SEV-3
      log:
        output: /var/log/teleport-datadog.log
        severity: DEBUG
    asserts:
      - matchSnapshot: {}

  - it: should not contain annotations when not defined
    asserts:
      - isNull:
          path: metadata.annotations

  - it: should contain annotations when defined
    set:
      annotations:
        config:
          keyA: valA
          keyB: valB
    asserts:
      - equal:
          path: metadata.annotations
          value:
            keyA: valA
            keyB: valB
