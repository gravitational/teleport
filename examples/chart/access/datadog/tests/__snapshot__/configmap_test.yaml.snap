should match the snapshot:
  1: |
    apiVersion: v1
    data:
      teleport-datadog.toml: |
        [teleport]
        addr = "teleport.example.com:1234"
        identity = "/var/lib/teleport/plugins/datadog/teleport-identity/auth_id"
        refresh_identity = true

        [datadog]
        api_endpoint = "https://api.datadoghq.com"
        api_key = "/var/lib/teleport/plugins/datadog/datadog-api-key"
        application_key = "/var/lib/teleport/plugins/datadog/datadog-application-key"
        severity = "SEV-3"

        [role_to_recipients]
        "*" = ["admin@example.com"]

        [log]
        output = "/var/log/teleport-datadog.log"
        severity = "DEBUG"
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: teleport-plugin-datadog
        app.kubernetes.io/version: 17.0.0-alpha.5
        helm.sh/chart: teleport-plugin-datadog-17.0.0-alpha.5
      name: RELEASE-NAME-teleport-plugin-datadog
